CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(yc-libvhost-server)
+
+if(UNIX AND NOT APPLE)
+  set(LINUX TRUE)
+endif()
+
+if(NOT LINUX)
+  message(FATAL_ERROR "Unsupported platform")
+endif()
+
+SET(libvhost_log_verbosity "LOG_INFO" CACHE STRING "Libvhost log verbosity")
+message("Compiler ${CMAKE_C_COMPILER}")
+message("Libvhost log verbosity: ${libvhost_log_verbosity}")
+
+add_library(vhost-server)
+add_compile_definitions(_GNU_SOURCE LOG_VERBOSITY=${libvhost_log_verbosity})
+target_compile_options(vhost-server PRIVATE
+  -Wall
+  -Werror
+  -Wextra
+  -Wno-unused-parameter
+  -g
+  -O2
+
+  -Wno-error=unused-value
+  -Wno-error=unused-result
+  -Wno-error=strict-aliasing
+)
```

Comment:
Last time I checked these 3 were only needed for GCC, so we can put them under an if

Reply:
Fixed. Also added compiler-specific flags that enforce coding standards to match meson config.